ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/dialogs/PluginSelectionDialog.java

```diff
@@ -298,4 +305,22 @@ protected void createButtonsForButtonBar(Composite parent) {
 		createButton(parent, IDialogConstants.CANCEL_ID, IDialogConstants.CANCEL_LABEL, false);
 	}
 
-}
+	@Override
+	protected void updateButtonsEnableState(IStatus status) {
+		super.updateButtonsEnableState(status);
+		Button okButton = getOkButton();
+		StructuredSelection currentSelection = super.getSelectedItems();
+		HashMap<String, Boolean> existingImports = PluginSelectionDialog
+				.getExistingImports(PDEPlugin.getDefault().getLabelProvider().getCurrentPluginModel(), false);
+		if (!currentSelection.isEmpty())
+			okButton.setEnabled(false);
+		for (Object selection : currentSelection) {
+			if (selection instanceof IPluginModelBase
+					&& !(existingImports.keySet().contains(((IPluginModelBase) selection).getPluginBase().getId()))) {
```

Comment:
Please use pattern matching for instanceof in all changed code and if applicable.
```suggestion
			if (selection instanceof IPluginModelBase pluginModel && !(existingImports.keySet().contains(pluginModel).getPluginBase().getId()))) {
```
Please also reformat the line, not sure if it fits in one line now.